log-collector/src/lib.rs

```diff
@@ -1,15 +1,25 @@
 pub use log;
-use std::{cell::RefCell, rc::Rc};
-
+#[cfg(any(feature = "dev-context-only-utils", feature = "svm-internal"))]
+use {
+    qualifier_attr::qualifiers,
+    std::{cell::RefCell, rc::Rc},
+};
+#[cfg(any(feature = "dev-context-only-utils", feature = "svm-internal"))]
 const LOG_MESSAGES_BYTES_LIMIT: usize = 10 * 1000;
 
-pub struct LogCollector {
-    pub messages: Vec<String>,
-    pub bytes_written: usize,
-    pub bytes_limit: Option<usize>,
-    pub limit_warning: bool,
+#[cfg(any(feature = "dev-context-only-utils", feature = "svm-internal"))]
+#[cfg_attr(
+    any(feature = "dev-context-only-utils", feature = "svm-internal"),
+    qualifiers(pub)
+)]
+struct LogCollector {
```

Comment:
litesvm needs the LogCollector type because it calls the `process_message` function from `solana-svm`.

It also needs to be able to read and write the `bytes_limit` field